Pull in classes from Stacks-Editor

b-kelly opened the pull request:
This PR adds in a new `is-readonly` input class and abstracts out the button `is-loading` class to be used anywhere. I've added examples / docs for both items, as well as tested in both dark and highcontrast modes.

Ideally, this would be multiple PRs since they're individual features, but we can merge w/o squashing if you'd like each commit to appear in the changelog.

b-kelly committed:
feat(inputs): add .is-readonly input classes

b-kelly committed:
feat(spinner): abstract out btn.is-spinner into a standalone class

dancormier commented on lib/css/components/inputs.less:
```diff
@@ -462,21 +462,23 @@ fieldset {
 .s-input,
 .s-textarea,
 .s-select > select {
```
> **Note**
> The `readonly` attribute does not apply to `select` elements. This style was pre-existing and won't hurt anything to include, but I was a little confused, so I'm making a note.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly):
> The attribute is not supported or relevant to [`<select>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) 

The [section detailing the differences between `readonly` and `disabled`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#attribute_interactions) is worth a read too:
> The difference between [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled) and readonly is that read-only controls can still function and are still focusable, whereas disabled controls can not receive focus and are not submitted with the form and generally do not function as controls until they are enabled.

b-kelly replied:
I can also drop a note in a comment in the file. I did test the readonly styles on selects (because why not), but they looked very "meh", so I dropped them from the `.is-readonly` rule below since they aren't technically supported anyways. Good callout

dancormier commented on docs/_data/spinner.json:
```diff
@@ -24,6 +24,11 @@
+    },
+    {
+      "class": ".is-loading",
+      "applies": "Any element",
```
I tried this on a bunch of miscellaneous components and it works _to varying degrees_. If an element already has pseudo-elements, this class is likely to overwrite (or at least append) them. I found this with `.s-select`. In many other cases, the loading spinner isn't quite aligned the way you'd expect.

Anyone applying this class will need to use it thoughtfully and I can see a future where someone comes to the Stacks team in need of a component-specific application. That being said, I think this is progress.

b-kelly replied:
Hmm... that's a good call. I'll add some notes to the docs. Instead of "any element" maybe something like "any text display element"? That's pretty confusing though. I'll also drop a note to prefer using the spinner markup if possible.

Alternatively, we could leave this relatively undocumented and just officially support the class on `s-btn`, but I'd like to avoid that scenario if we can.

dancormier replied:
I updated the documentation in https://github.com/StackExchange/Stacks/pull/1040/commits/ac44000f502cdd247d342c7d86bde920fad81528. I'm not 100% satisfied with the wording, but I don't think we need to spend too much more time on it.

dancormier reviewed (approved):
I left a couple of notes that require no action. This is 🦐 status. Thanks!

> Ideally, this would be multiple PRs since they're individual features, but we can merge w/o squashing if you'd like each commit to appear in the changelog.

Merge w/o squash would be nice but I'm [not sure GitHub will allow that](https://github.com/community/community/discussions/10809), so a single commit is fine in this case if it's trouble.

dancormier committed:
Merge branch 'develop' into bkelly/stacks-editor-classes

dancormier committed:
docs(inputs): add comment on select readonly attr

addresses https://github.com/StackExchange/Stacks/pull/1040#discussion_r931144086

dancormier committed:
docs(spinner): update is-loading documentation

addresses https://github.com/StackExchange/Stacks/pull/1040#discussion_r931158632

dancormier committed:
Merge branch 'develop' into bkelly/stacks-editor-classes
